pkg/server/registry/images/tag.go

```diff
@@ -58,9 +59,25 @@ func (t *TagStrategy) ImageTag(ctx context.Context, namespace, imageName string,
 	if err != nil {
 		return nil, err
 	}
+	duplicateList, err := checkDuplicateTags(image, imageList, &tags)
+	if err != nil {
+		return nil, err
+	}
+	for _, duplicateTag := range duplicateList {
+		err = t.client.Update(ctx, &duplicateTag)
+		if err != nil {
+			return image, err
+		}
+	}
 
+	image.Tags = append(image.Tags, normalizeTags(tags)...)
+	return image, t.client.Update(ctx, image)
+}
+
+func checkDuplicateTags(image *apiv1.Image, imageList *apiv1.ImageList, tags *[]string) ([]apiv1.Image, error) {
```

Comment:
Why is this a `*[]string`?  It's generally bad form to take a pointer to a slice unless you are doing something very specific and tricky.

Reply:
Oh, I see on L118.  That's not good, too obscure to modify a slice like that.  What is better is just collect all the tags up front so something like

```golang
set := sets.NewString()
for _, tag := range append(image.Tags, tags...) {
   set.Insert(normalize(tag))
}

// then remove tags from other images that match

image.Tags = set.List()
```

Reply:
Should be easier to follow now